### PR TITLE
New Sign In Screen

### PR DESCRIPTION
**Pull request by sevvaleygul0**
## What

- New screen is created for sign in.

## Demo

https://user-images.githubusercontent.com/56435142/120082038-56135880-c0c9-11eb-92db-e9080d368b30.mov

Screen Shot 2021-05-29 at 22 00 00

### CODE REVIEW

File: lib/SocialLoginScreen.tsx [hunk 1]
```diff
@@ -354,17 +381,64 @@ export default class SocialLoginScreen extends React.PureComponent<
...
+            emailTextFieldStyle={this.props.emailTextFieldStyle}
+            signinButtonContainer={this.props.signinButtonContainer}
+            onUserNameChangeText={this.props.onNewUsernameChangeText}
+            onPasswordChangeText={this.props.onNewPasswordChangeText}
+            onloginTextPress={() =>
+              this.setState({ signUpScreenVisibility: false })
+            }
+            onSigninPress={this.props.onSigninPress}
+            onEmailChangeText={this.props.onNewEmailChangeText}
+          />
```
Comment (kuraydev) [inline lib/SocialLoginScreen.tsx]: There are too many repetitions for "this.props.". You can use de-construction

File: lib/SocialLoginScreen.tsx [hunk 2]
```diff
@@ -354,17 +381,64 @@ export default class SocialLoginScreen extends React.PureComponent<
     );
   };
 
+  renderScreenControl = () => {
+    if (this.state.signUpScreenVisibility) {
+      return (
+        <SafeAreaView style={styles.container}>
+          <SignUpScreen
+            loginText={this.props.loginText}
```
Comment (kuraydev) [inline lib/SocialLoginScreen.tsx]: Can't we just prop drilling instead of set the all of the props.

File: lib/signUpScreen/SignUpScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,206 @@
+import * as React from "react";
```
Comment (kuraydev) [inline lib/signUpScreen/SignUpScreen.tsx:1]: I think we do not need a username here, we need a password and re-password here

File: lib/signUpScreen/SignUpScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+interface IState {}
+
+export default class SignUpScreen extends React.PureComponent<
+  ISocialLoginProps,
+  IState
+> {
+  renderHeader = () => {
+    const {
+      loginText = "LOG IN",
```
Comment (kuraydev) [inline lib/signUpScreen/SignUpScreen.tsx]: This is a signup screen, can you change the loginText and all of them to signUp.. Also "LOG IN" should change to "SIGN UP"

File: lib/signUpScreen/SignUpScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+  renderSigninTitle = () => {
+    const { signinTitleText = "Sign In", signinTextStyle } = this.props;
+    return (
+      <View style={styles.signinTitleContainer}>
+        <Text style={[styles.signinTextStyle, signinTextStyle]}>
+          {signinTitleText}
+        </Text>
+      </View>
+    );
+  };
```
Comment (kuraydev) [inline lib/signUpScreen/SignUpScreen.tsx]: This is the signup screen, can you change the "signin" namespace to "signup".

File: lib/signUpScreen/SignUpScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+          resizeMode="contain"
+          source={rightTopAssetImageSource}
+          style={styles.rightTopAssetImageStyle}
+        />
+      </View>
+    );
+  };
+
+  renderSigninTitle = () => {
+    const { signinTitleText = "Sign In", signinTextStyle } = this.props;
```
Comment (kuraydev) [inline lib/signUpScreen/SignUpScreen.tsx]: This string should be "Sign Up"

**Review by kuraydev — CHANGES_REQUESTED** [review]
There should be not local lib in the example

**Comment by sevvaleygul0** [comment]
https://user-images.githubusercontent.com/56435142/120279869-a18f5780-c2bf-11eb-965b-12cd01448a4d.mov

**Comment by kuraydev** [comment]
>  Screen.Recording.2021-06-01.at.09.55.40.mov

The second password is not secure text. Can you fix it?

**Review by kuraydev — DISMISSED** [review]

**Review by kuraydev — APPROVED** [review]

**Pull request merged**